profile/src/main/java/com/scottlogic/deg/profile/reader/JsonProfileReader.java

```diff
@@ -73,51 +68,25 @@ public Profile read(String profileJson) {
             throw new InvalidProfileException("Profile is invalid: 'fields' have not been defined.");
         if (profileDTO.rules == null)
             throw new InvalidProfileException("Profile is invalid: 'rules' have not been defined.");
-
-        List<Field> fields = profileDTO.fields.stream()
-                .map(fieldDTO -> new Field(
-                    fieldDTO.name,
-                    fieldDTO.type.getFieldType(),
-                    fieldDTO.unique,
-                    getFormatting(fieldDTO),
-                    false))
-                .collect(Collectors.toList());
-
-        List<Field> inMapFields = profileDTO.rules.stream()
-            .flatMap(ruleDTO -> ruleDTO.constraints.stream())
-            .flatMap(constraintDTO -> getInMapConstraints(profileDTO).stream())
-            .distinct()
-            .map(file -> new Field(file, FieldType.NUMERIC, false, null, true)
-            ).collect(Collectors.toList());
-
-        fields.addAll(inMapFields);
-        ProfileFields profileFields = new ProfileFields(fields);
+        Collection<Field> fieldCollection = new ArrayList<>();
+        fieldCollection.addAll(profileDTO.fields.stream().map(this::createField).collect(Collectors.toList()));
+        fieldCollection.addAll(profileDTO.rules.stream().map(this::createInMapFields).flatMap(Collection::stream).collect(Collectors.toList()));
```

Comment:
In cases like this, I would prefer to see the stream split across lines per every method invocation. I find it makes it easier to read what is going on, as the flow of separate actions is vertical, and also it allows us to stay within a page width.

`profileDTO.rules.stream().map(this::createInMapFields).flatMap(Collection::stream).collect(Collectors.toList())`
vs.
`profileDTO.rules.stream()`
`  .map(this::createInMapFields)`
`  .flatMap(Collection::stream)`
`  .collect(Collectors.toList())`